### DISCUSSION TITLE
Make `value` non-optional for payable functions

### DISCUSSION THREAD

**Discussion by fubhy** [wevm/viem#372] [Idea / Feature Request] [answered]
I think it would be good to make `value` non-optional for payable functions, so you are forced to set it explicitly. Otherwise, it's easy to forget.

**Comment by tmm** [comment, marked as answer]
Agree! This is the default behavior in wagmi. Should have in viem as well. Welcome to create a PR! (Should add a test too so it doesn't get lost in the future as well.)

**Reply by fubhy** [reply]
Will do. Likely tomorrow though. Family time!

**Reply by tmm** [reply]
https://github.com/wagmi-dev/viem/pull/397

**Reply by fubhy** [reply]
Whoops. Forgot about this. Thanks <3 :-)
